### PR TITLE
adds traject webhook tasks

### PR DESCRIPTION
**Pull request by whereismyjetpack**
This is completely experimental 

When deployed with argo workflows, we inject a TRAJECT_WEBHOOK_URL environment variable into the pod. If that variable is present, we display these helper rake tasks 

`iterate_and_import` - uses solr_manager to create a collection, and then fire off a traject webhook to index into this collection 

`clear_hourlies` allows us to run the clear_hourlies traject rake task from the webhook.

`hourlies` allows us to then catch up on hourlies from the webhook as well.

### CODE REVIEW

File: lib/tasks/traject.rake [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+require 'psulib_blacklight/solr_manager'
+require 'psulib_blacklight/data_manager'
+
+require 'net/http'
+
+namespace :traject do
+  # We only run this task in kubernetes. Hide it if the WEBHOOK_URL is not set
+  if ENV['TRAJECT_WEBHOOK_URL']
+    desc 'Creates a new collection, the fires off traject job to index into that collection'
```
Comment (banukutlu) [inline lib/tasks/traject.rake:11]:
```suggestion
    desc 'Creates a new collection, then fires off traject job to index into that collection'
```

File: lib/tasks/traject.rake [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    task clear_hourlies: :environment do
+      traject_webhook_url = ENV['TRAJECT_WEBHOOK_URL']
+      uri = URI("http://#{traject_webhook_url}/rake")
+
+      req = Net::HTTP::Post.new(uri, 'Content-Type' => 'application/json')
+      req.body = { command: 'traject:clear_hourlies' }.to_json
+      res = Net::HTTP.start(uri.hostname, uri.port) do |http|
+        http.request(req)
+      end
+      puts "Webhook repsonded with Status: #{res.code} #{res.message}"
```
Comment (banukutlu) [inline lib/tasks/traject.rake:41]:
```suggestion
      puts "Webhook responded with Status: #{res.code} #{res.message}"
```

File: lib/tasks/traject.rake [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+      puts "Asking Traject to index #{path} to #{collection}"
+
+      req = Net::HTTP::Post.new(uri, 'Content-Type' => 'application/json')
+      req.body = { collection: collection, path: path }.to_json
+      res = Net::HTTP.start(uri.hostname, uri.port) do |http|
+        http.request(req)
+      end
+
+      puts "Webhook repsonded with Status: #{res.code} #{res.message}"
+      puts "When Traject is done, don't forget to process dailies and flip the collection"
```
Comment (banukutlu) [inline lib/tasks/traject.rake:28]:
```suggestion
      puts "When Traject is done, don't forget to process incrementals and flip the collection"
```

File: lib/tasks/traject.rake [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+      uri = URI("http://#{traject_webhook_url}/hourlies")
+      collection = args['collection'] || solr_manager.last_incremented_collection
+
+      puts "Asking Traject to run hourlies on #{collection}"
+      req = Net::HTTP::Post.new(uri, 'Content-Type' => 'application/json')
+      req.body = { collection: collection }.to_json
+      res = Net::HTTP.start(uri.hostname, uri.port) do |http|
+        http.request(req)
+      end
+      puts "Webhook repsonded with Status: #{res.code} #{res.message}"
```
Comment (banukutlu) [inline lib/tasks/traject.rake:56]:
```suggestion
      puts "Webhook responded with Status: #{res.code} #{res.message}"
```

**Review by banukutlu — CHANGES_REQUESTED** [review]
This looks great to me, just minor text typos which has nothing to do real work :)

File: lib/tasks/traject.rake [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+require 'net/http'
+
+namespace :traject do
+  # We only run this task in kubernetes. Hide it if the WEBHOOK_URL is not set
+  if ENV['TRAJECT_WEBHOOK_URL']
+    desc 'Creates a new collection, then fires off traject job to index into that collection'
+    task :iterate_and_import, [:path] => [:environment] do |_task, args|
+      path = args['path'] || 'full_extracts'
+      traject_webhook_url = ENV['TRAJECT_WEBHOOK_URL'] || raise('TRAJECT_WEBHOOK_URL is missing')
```
Comment (nicktiberi) [inline lib/tasks/traject.rake:14]: I don't think it's possible for `TRAJECT_WEBHOOK_URL` to be missing due to the check at line 10, right?

Comment (whereismyjetpack) [reply lib/tasks/traject.rake:14]: correct. line 10 was added later, when I realized that it's not terribly useful to have these tasks outside of k8s

**Review by nicktiberi — APPROVED** [review]
LGTM! Just need to fix that dang rubocop trailing space issue and it is good to go from my perspective. 

Thanks so much for putting this together! It's like magic 🪄

**Review by nicktiberi — APPROVED** [review]
🚀

**Review by banukutlu — APPROVED** [review]
🚀  looks great!

**Pull request merged**